crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -53,8 +54,15 @@ pub async fn check_transaction_input(
     // Check for type arguements and ensure they exist
     // ignore the returned objects because all we care about is that they exist for now in order to
     // prevent more forks
-    let type_argument_packages = transaction.type_argument_packages();
-    store.check_input_objects(&type_argument_packages)?;
+    let type_arg_obj_kinds = transaction.type_argument_packages();
+    let type_arg_objs = store.check_input_objects(&type_arg_obj_kinds)?;
+
+    // This check would prevent users to submit a transaction with non-move-package
+    // as type arguments.
+    for (object_kind, object) in type_arg_obj_kinds.into_iter().zip(type_arg_objs) {
```

Comment:
Maybe type arg objs should just be added to `input_objects` and run the `check_objects()` call below. But this seems fine.